packages/sdk/src/cross-chain-messenger.ts

```
@@ -1212,29 +1258,85 @@ export class CrossChainMessenger {
+      // there's a devex edge case where this message appears to give back a valid proposal that
+      // ends up reverting inside of proveMessage. At least this is safe for users but not ideal
+      // for developers and we should work out the simplest way to fix it. Main blocker is that
+      // verifying the proposal may require access to an archive node.
+
+      // If there's no match then we can't prove the message to the portal.
+      if (!match) {
         return null
-      } else {
-        throw err
       }
-    }
 
-    // Now pull the proposal out given the output index. Should always work as long as the above
-    // codepath completed successfully.
-    const proposal = await this.contracts.l1.L2OutputOracle.getL2Output(
-      l2OutputIndex
-    )
+      // Put the result into the same format as the old logic for now to reduce added code.
+      l2OutputIndex = match.index
+      proposal = {
+        outputRoot: match.rootClaim,
+        timestamp: match.timestamp,
+        l2BlockNumber: match.l2BlockNumber,
+      }
+    } else {
+      // Try to find the output index that corresponds to the block number attached to the message.
+      // We'll explicitly handle "cannot get output" errors as a null return value, but anything else
+      // needs to get thrown. Might need to revisit this in the future to be a little more robust
+      // when connected to RPCs that don't return nice error messages.
+      try {
+        l2OutputIndex =
+          await this.contracts.l1.L2OutputOracle.getL2OutputIndexAfter(
+            resolved.blockNumber
+          )
+      } catch (err) {
+        if (err.message.includes('L2OutputOracle: cannot get output')) {
+          return null
+        } else {
+          throw err
+        }
+      }
+
+      // Now pull the proposal out given the output index. Should always work as long as the above
+      // codepath completed successfully.
+      proposal = await this.contracts.l1.L2OutputOracle.getL2Output(
```

I think this is right but just in case asking the question (since not typesafe). Can this be undefined? And if it is undefined do we want to throw an error?

---

Only case where this could possibly be undefined is if the output is deleted, in that case you'll just get a runtime error. Not ideal but whatever. Also this is for the legacy codepath that will no longer be relevant after FPAC ships.

---

Sgtm